### PR TITLE
Avoid using templates in BufferManager.

### PR DESCRIPTION
**Pull request by S-Dafarra**
With this PR I completely removed the use of templates for ``BufferManager`` since the default template was in any case causing issues.

I have also fixed a small discrepancy in the definition of the dimensions of cell and struct arrays.

### CODE REVIEW

**Comment by S-Dafarra** [comment]
cc @GiulioRomualdi @traversaro

File: README.md [hunk 1]
```diff
@@ -364,7 +363,7 @@ ans =
 
   struct with fields:
 
-              data: {3×1 cell}
+              data: {1×3 cell}
```
Comment (Nicogene) [inline README.md:366]: Why did you change it?

**Review by Nicogene — CHANGES_REQUESTED** [review]
Could you please add [this test](https://github.com/robotology/robometry/commit/fac3227c47aefc60e89442ee75f1365ead6d8057) for checking the issue this PR is addressing?

File: README.md (hunk 1 above)
Comment (S-Dafarra) [reply README.md:366]: See https://github.com/robotology/robometry/pull/175/commits/a6309eedc8419607dcb68f870bd87fd46eebe14a, I realized that for cell and struct arrays I was concatenating on the first dimension. This was causing differences with the ``dimensions`` vector.

You can actually see this mismatch in the previous version of the README: https://github.com/robotology/robometry/blob/14310e9a80c75eba987d647f02a91afe9f0c0512/README.md?plain=1#L367-L368

Comment (Nicogene) [reply README.md:366]: Aaa I see

**Review by Nicogene — APPROVED** [review]
LGTM!

Is it ok for you @GiulioRomualdi @traversaro ?

**Review by traversaro — APPROVED** [review]

**Pull request merged**